### PR TITLE
Allow functions for assert and refute messages

### PR DESCRIPTION
**Pull request by mroderick**
This allows for dynamic assert and refute messages, which is required to improve combined assertions like `calledOnceWith` in `@sinonjs/refere-sinon`. See https://github.com/sinonjs/referee-sinon/pull/113

#### How to verify - mandatory

1. Check out this branch
2. `npm install`
3. <your-steps-here>

#### Checklist for author

- [ ] `npm run lint` passes
- [ ] References to standard library functions are [cached](https://github.com/sinonjs/sinon/pull/1523).

### CODE REVIEW

File: lib/create-add.js [hunk 1]
```diff
@@ -31,19 +31,35 @@ function verifyArguments(name, options) {
...
   }
 
-  if (
-    typeof options.assertMessage !== "string" ||
-    options.assertMessage.length === 0
-  ) {
-    throw new TypeError("'assertMessage' option must be a non-empty String");
+  ["assertMessage", "refuteMessage"].forEach((m) => {
+    if (!isValidMessageOption(options[m])) {
+      throw new TypeError(`'${m}' option must be a non-empty String`);
```
Comment (mantoni) [inline lib/create-add.js:36]: > option must be a non-empty string or a function

File: lib/create-add.js [hunk 2]
```diff
@@ -31,19 +31,35 @@ function verifyArguments(name, options) {
...
+    }
+  });
+}
+
+/**
+ * returns true when message option is considered valie,
```
Comment (mantoni) [inline lib/create-add.js:42]: `valid` typo

File: lib/create-add.js [hunk 3]
```diff
@@ -31,19 +31,35 @@ function verifyArguments(name, options) {
...
+ * i.e. when it is a function or a non-empty string
+ *
+ * @private
+ * @param  {*}  message [description]
```
Comment (mantoni) [inline lib/create-add.js:46]: Can we omit the `[description]` if no description is added?

**Review by mantoni — APPROVED** [review]
Just some minor comments 👍

File: lib/create-add.js (hunk 3 above)
Comment (mroderick) [reply lib/create-add.js]: Yes

**Review by hexeberlin — APPROVED** [review]

**Comment by mroderick** [comment]
This has been published as `@sinonjs/referee@9.1.0`

**Pull request merged**